### PR TITLE
Distributed K-Means: fix for step #2

### PR DESCRIPTION
**Pull request by amgrigoriev**

### CODE REVIEW

File: cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels_distr_steps.cl [hunk 1]
```diff
@@ -45,9 +45,9 @@ DECLARE_SOURCE(
         const int local_id                  = get_local_id(1);
         const int oldN                      = partialCentroidsCounters[global_id];
         const int newN                      = cCounters[global_id];
-        const algorithmFPType oldContrib    = oldN * partialCentroids[global_id * P + local_id];
-        const algorithmFPType newContrib    = newN * centroids[global_id * P + local_id];
-        centroids[global_id * P + local_id] = (oldContrib + newContrib) / (oldN + newN);
+        const algorithmFPType oldContrib    = oldN > 0 ? oldN * partialCentroids[global_id * P + local_id] : 0.0;
```
Comment (KulikovNikita) [inline cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels_distr_steps.cl:48]:
```suggestion
        const algorithmFPType oldContrib    = oldN > 0 ? oldN * partialCentroids[global_id * P + local_id] : algorithmFPType(0);
```

File: cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels_distr_steps.cl [hunk 2]
```diff
@@ -45,9 +45,9 @@ DECLARE_SOURCE(
...
+        const algorithmFPType newContrib    = newN > 0 ? newN * centroids[global_id * P + local_id] : 0.0;
```
Comment (KulikovNikita) [inline cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels_distr_steps.cl:49]:
```suggestion
        const algorithmFPType newContrib    = newN > 0 ? newN * centroids[global_id * P + local_id] : algorithmFPType(0);
```

File: cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels_distr_steps.cl [hunk 3]
```diff
@@ -45,9 +45,9 @@ DECLARE_SOURCE(
...
+        centroids[global_id * P + local_id] = (oldN + newN > 0) ? (oldContrib + newContrib) / (oldN + newN) : 0.0;
```
Comment (KulikovNikita) [inline cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels_distr_steps.cl:50]:
```suggestion
        centroids[global_id * P + local_id] = (oldN + newN > 0) ? (oldContrib + newContrib) / (oldN + newN) : algorithmFPType(0);
```

**Review by KulikovNikita — COMMENTED** [review]
Please note `0.0` is usually interpreted by compiler as `double`. It may cause problems on some systems

File: cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels_distr_steps.cl (hunk 1 above)
Comment (makart19) [inline cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels_distr_steps.cl:48]:
```suggestion
        const algorithmFPType oldContrib    = oldN > 0 ? oldN * partialCentroids[global_id * P + local_id] : algorithmFPType(0);
```

Comment (makart19) [inline cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels_distr_steps.cl:48]:
```suggestion
        const algorithmFPType oldContrib    = oldN > 0 ? oldN * partialCentroids[global_id * P + local_id] : (algorithmFPType)0;
```

File: cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels_distr_steps.cl (hunk 2 above)
Comment (makart19) [inline cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels_distr_steps.cl:49]:
```suggestion
        const algorithmFPType newContrib    = newN > 0 ? newN * centroids[global_id * P + local_id] : (algorithmFPType)0;
```

File: cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels_distr_steps.cl (hunk 3 above)
Comment (makart19) [inline cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels_distr_steps.cl:50]:
```suggestion
        centroids[global_id * P + local_id] = (oldN + newN > 0) ? (oldContrib + newContrib) / (oldN + newN) : (algorithmFPType)0;
```

**Review by makart19 — APPROVED** [review]

**Pull request merged**